Functions/Solution.cs

```diff
@@ -15,74 +16,267 @@ internal class Solution
-        {
-            /*
-             * TODO: Write code here
-             */
+        {   
+            Regex rgx = new Regex("[^a-zA-Z]");
+            str = rgx.Replace(str, "");
+            string sentence = str.ToLower();
+            int n = sentence.Length - 1;
+            bool isPalindrome = true;
+            int i = 0;
+            while(i < n-i)
+            {
+                if (sentence[i] != sentence[n-i])
+                {
+                    isPalindrome = false;
+                    break;
+                }
+                i++;
+            }
+            return isPalindrome;
         }
 
         public static int Assignment7(int[] nums)
         {
-            /*
-             * TODO: Write code here
-             */
+            var histogram = new Dictionary<int, int>();
+            foreach(int num in nums)
+            {
+                if (histogram.ContainsKey(num))
+                    histogram[num]++;
+                else
+                    histogram[num] = 1;
+            }
+            foreach(KeyValuePair<int, int> entry in histogram)
+            {
+                if (entry.Value == 1)
+                    return entry.Key;
+            }
+            Console.WriteLine("No number in array with only 1 input");
+            return -1;
         }
 
+        
         public static void Assignment8(bool[,] board, int rows, int cols)
-        {
-            /*
-             * TODO: Write code here
-             */
+        {            
+            bool[,] newBoard = new bool[rows, cols];
+            for(int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < cols; j++)
+                {
+                    int numAliveNeighs = GameOfLife.calcAliveNeighs(board, i, j, rows, cols);
+                    if (board[i, j])
+                    {
+                        switch(numAliveNeighs)
+                        {
+                            case 0: case 1: case 4:
+                                newBoard[i, j] = false;
+                                break;
+                            case 2: case 3:
+                                newBoard[i, j] = true;
+                                break;
+                            default:
+                                Console.WriteLine("Shouldn't get here.");
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        if (numAliveNeighs == 3)
+                            newBoard[i, j] = true;
+                        else
+                            newBoard[i, j] = false;
+                    }
+                }
+                
+            }
+            // print output matrix
+            for (int i = 0; i < rows; i++)
+            {
+                for (int j = 0; j < cols; j++)
+                {
+                    Console.Write(string.Format("{0} ", newBoard[i, j]));
```

Comment:
Use [string templating](https://stackoverflow.com/questions/733378/whats-a-good-way-of-doing-string-templating-in-net).